mass migration not working when using database postgres

### Steps to reproduce

1. go to Browse in the left sidebar
2. go to Migrate tab
3. choose any source, and then postgres error showing up

### Expected behavior

I should be able to mass migrate source.

### Actual behavior

this error showed up:

<img width="1575" height="887" alt="Image" src="https://github.com/user-attachments/assets/71f2d4f2-4ff3-4bed-b0ec-8acd682984ed" />

### Suwayomi-Server version

v2.2.2149

### Used database

PostgreSQL

### Server operating system

Ubuntu 24.04 LTS

### Server Desktop Environment

_No response_

### Server JVM version

openjdk version "25.0.3" 2026-04-21

### Used client name

Suwayomi - WebUI

### Client version

r3180

### Used web browser

Brave 1.90.122

### Client operating system

Ubuntu 24.04 LTS

### Other details

i’m using docker compose with the image ghcr.io/suwayomi/suwayomi-server:preview, so the environment should be standard.

the database image is using postgres:17.

here are the error logs i get from docker compose logs:
```
suwayomi  | 14:40:54.102 [DefaultDispatcher-worker-3] ERROR suwayomi.tachidesk.graphql.server.TachideskGraphQLServer -- GraphQL execution failed due to
suwayomi  | org.jetbrains.exposed.v1.exceptions.ExposedSQLException: org.postgresql.util.PSQLException: ERROR: SELECT DISTINCT ON expressions must match initial ORDER BY expressions
suwayomi  |   Position: 43
suwayomi  | SQL: [Failed on expanding args for SELECT: org.jetbrains.exposed.v1.jdbc.Query@47eab2ca]
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.statements.BlockingExecutableKt.executeIn(BlockingExecutable.kt:158)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.JdbcTransaction.exec$lambda$3(JdbcTransaction.kt:237)
suwayomi  |     at org.jetbrains.exposed.v1.core.transactions.TransactionsKt.withThreadLocalTransaction(Transactions.kt:47)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.JdbcTransaction.exec(JdbcTransaction.kt:233)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.JdbcTransaction.execQuery(JdbcTransaction.kt:291)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.Query.count(Query.kt:255)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.Query.count(Query.kt:248)
suwayomi  |     at suwayomi.tachidesk.graphql.queries.MangaQuery.mangas$lambda$0(MangaQuery.kt:266)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.transactions.TransactionsKt.inTopLevelTransaction$lambda$0(Transactions.kt:193)
suwayomi  |     at org.jetbrains.exposed.v1.core.transactions.TransactionsKt.withThreadLocalTransaction(Transactions.kt:53)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.transactions.TransactionsKt.inTopLevelTransaction(Transactions.kt:189)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.transactions.TransactionsKt.transaction(Transactions.kt:134)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.transactions.TransactionsKt.transaction$default(Transactions.kt:107)
suwayomi  |     at suwayomi.tachidesk.graphql.queries.MangaQuery.mangas(MangaQuery.kt:245)
suwayomi  |     at suwayomi.tachidesk.graphql.queries.MangaQuery.mangas$default(MangaQuery.kt:224)
suwayomi  |     at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
suwayomi  |     at java.base/java.lang.reflect.Method.invoke(Unknown Source)
suwayomi  |     at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97)
suwayomi  |     at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Static.call(CallerImpl.kt:106)
suwayomi  |     at kotlin.reflect.jvm.internal.ReflectKCallableKt.callDefaultMethod(ReflectKCallable.kt:161)
suwayomi  |     at kotlin.reflect.jvm.internal.ReflectKCallableImpl.default$callBy(ReflectKCallableImpl.kt:58)
suwayomi  |     at kotlin.reflect.jvm.internal.ReflectKCallableImpl.callBy(ReflectKCallableImpl.kt:8)
suwayomi  |     at com.expediagroup.graphql.generator.execution.FunctionDataFetcher.runBlockingFunction(FunctionDataFetcher.kt:127)
suwayomi  |     at com.expediagroup.graphql.generator.execution.FunctionDataFetcher.get(FunctionDataFetcher.kt:63)
suwayomi  |     at suwayomi.tachidesk.graphql.directives.RequireAuthDirectiveWiring.onField$lambda$0(RequireAuthDirectiveWiring.kt:44)
suwayomi  |     at graphql.execution.ExecutionStrategy.invokeDataFetcher(ExecutionStrategy.java:512)
suwayomi  |     at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:466)
suwayomi  |     at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:404)
suwayomi  |     at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:360)
suwayomi  |     at graphql.execution.ExecutionStrategy.getAsyncFieldValueInfo(ExecutionStrategy.java:325)
suwayomi  |     at graphql.execution.AsyncExecutionStrategy.execute(AsyncExecutionStrategy.java:58)
suwayomi  |     at graphql.execution.Execution.executeOperation(Execution.java:220)
suwayomi  |     at graphql.execution.Execution.execute(Execution.java:139)
suwayomi  |     at graphql.GraphQL.execute(GraphQL.java:620)
suwayomi  |     at graphql.GraphQL.lambda$parseValidateAndExecute$5(GraphQL.java:549)
suwayomi  |     at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(Unknown Source)
suwayomi  |     at java.base/java.util.concurrent.CompletableFuture.thenCompose(Unknown Source)
suwayomi  |     at graphql.EngineRunningState.compose(EngineRunningState.java:91)
suwayomi  |     at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:544)
suwayomi  |     at graphql.GraphQL.lambda$executeAsync$1(GraphQL.java:504)
suwayomi  |     at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(Unknown Source)
suwayomi  |     at java.base/java.util.concurrent.CompletableFuture.thenCompose(Unknown Source)
suwayomi  |     at graphql.EngineRunningState.compose(EngineRunningState.java:91)
suwayomi  |     at graphql.GraphQL.lambda$executeAsync$2(GraphQL.java:493)
suwayomi  |     at graphql.EngineRunningState.engineRun(EngineRunningState.java:202)
suwayomi  |     at graphql.GraphQL.executeAsync(GraphQL.java:485)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.execute(GraphQLRequestHandler.kt:108)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest$suspendImpl(GraphQLRequestHandler.kt:85)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest(GraphQLRequestHandler.kt)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invokeSuspend(GraphQLServer.kt:62)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)
suwayomi  |     at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatched(Undispatched.kt:66)
suwayomi  |     at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
suwayomi  |     at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:848)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLServer.execute$suspendImpl(GraphQLServer.kt:49)
suwayomi  |     at com.expediagroup.graphql.server.execution.GraphQLServer.execute(GraphQLServer.kt)
suwayomi  |     at suwayomi.tachidesk.graphql.controller.GraphQLController$execute$1$1.invokeSuspend(GraphQLController.kt:26)
suwayomi  |     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
suwayomi  |     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
suwayomi  |     at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
suwayomi  |     at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
suwayomi  |     at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
suwayomi  |     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:798)
suwayomi  |     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
suwayomi  |     at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
suwayomi  | Caused by: org.postgresql.util.PSQLException: ERROR: SELECT DISTINCT ON expressions must match initial ORDER BY expressions
suwayomi  |   Position: 43
suwayomi  |     at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2904)
suwayomi  |     at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2589)
suwayomi  |     at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:431)
suwayomi  |     at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:534)
suwayomi  |     at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:437)
suwayomi  |     at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:196)
suwayomi  |     at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:139)
suwayomi  |     at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
suwayomi  |     at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.statements.jdbc.JdbcPreparedStatementImpl.executeQuery(JdbcPreparedStatementImpl.kt:53)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.Query.executeInternal(Query.kt:288)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.Query.executeInternal(Query.kt:15)
suwayomi  |     at org.jetbrains.exposed.v1.jdbc.statements.BlockingExecutableKt.executeIn(BlockingExecutable.kt:156)
suwayomi  |     ... 65 common frames omitted
```

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [x] I have written a short but informative title (ideally less than ~100 characters).
- [x] I have tried the troubleshooting guide described in [README.md](https://github.com/Suwayomi/Suwayomi-Server?tab=readme-ov-file#troubleshooting-and-support)
- [x] I have updated to the **[latest version](https://github.com/suwayomi/suwayomi-server/releases/latest)**.
- [x] I have filled out all of the requested information in this form, including specific version numbers.
- [x] I understand that **Suwayomi does not have or fix any extensions**, and I **will not receive help** for any issues related to sources or extensions.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - (**Extension**) Do not indicate an update is available when the extension is not installed
 - (**Chapter**) Fix losing chapter data on failed chapter list update
 - (**Chapter**) Fix database error when fetching chapter updates
+- (**Manga/API**) Fix "mangas" graphql query with active sorting and using a postgresql database (QUERY "mangas")
 
 ## [v2.2.2100] + [WebUI: v20260508.01] - 2026-05-08
 

--- a/server/src/main/kotlin/suwayomi/tachidesk/graphql/queries/MangaQuery.kt
+++ b/server/src/main/kotlin/suwayomi/tachidesk/graphql/queries/MangaQuery.kt
@@ -247,11 +247,18 @@ class MangaQuery {
                     MangaTable
                         .leftJoin(CategoryMangaTable)
                         .select(MangaTable.columns)
-                        .withDistinctOn(MangaTable.id)
+
+                val applySorting = order != null || orderBy != null || (last != null || before != null)
+
+                if (applySorting) {
+                    res.withDistinctOn(MangaTable.id to SortOrder.ASC)
+                } else {
+                    res.withDistinctOn(MangaTable.id)
+                }
 
                 res.applyOps(condition, filter)
 
-                if (order != null || orderBy != null || (last != null || before != null)) {
+                if (applySorting) {
                     val baseSort = listOf(MangaOrder(MangaOrderBy.ID, SortOrder.ASC))
                     val deprecatedSort = listOfNotNull(orderBy?.let { MangaOrder(orderBy, orderByType) })
                     val actualSort = (order.orEmpty() + deprecatedSort + baseSort)

